nixos/lib/make-options-doc/options-to-docbook.xsl

```diff
@@ -39,8 +39,8 @@
             concat('opt-',
               translate(
                 attr[@name = 'name']/string/@value,
-                '*&lt; >[]:',
-                '_______'
+                '*&lt; >[]:&quot;',
+                '________'
```

Comment:
This changes the identifiers, breaking inbound links. Could you remove the quotes instead of replacing them with underscores?

Reply:
Shouldn't that throw an error if `<xref linkend="" />` points to an unknown location?

Reply:
Then that would have been a problem before.
You'd have to define `options."" = mkOption ...` at the top level, which I don't think is something we have to worry about.

Reply:
> Then that would have been a problem before.

Yes. But considering that I can build the manual locally, doesn't that mean that there are no breaking links?

> You'd have to define options."" = mkOption ... at the top level, which I don't think is something we have to worry about.

Even though it's just a theoretical problem, having e.g. `settings."foo.bar"` and `settings.foo.bar` defined would cause a conflict when removing the quotes, that's why I kept them.

Or am I misunderstanding your point? :)

Reply:
ping @roberth

Reply:
I was more concerned about links from the web to the manual. Passing the internal validation will not detect such broken links.
It does appear that the impact is quite limited:

(`{+_+}` is git word diff syntax for "add underscore")

 - 138 ids relating to sr.ht such as `opt-services.sourcehut.settings.{+_+}sr.ht{+_+}.owner-name`
 - `opt-services.listmonk.database.settings.{+_+}app.notify_emails{+_+}`
 - `opt-services.listmonk.database.settings.{+_+}bounce.mailboxes{+_+}`
 - `opt-services.listmonk.database.settings.{+_+}privacy.domain_blocklist{+_+}`
 - `opt-services.listmonk.database.settings.{+_+}privacy.exportable{+_+}`
 - `opt-services.xserver.windowManager.{+_+}2bwm{+_+}.enable`

This seems like a small price to pay for a simpler and more injective id generating function.